### PR TITLE
feat: support initiating clients with projectId

### PR DESCRIPTION
**Pull request by alexanderczigler**

### CODE REVIEW

File: packages/pubsub/src/lib/client.ts [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
 
-export const pubsub = new PubSub()
+export type ClientConfig = {
+  projectId: string
+}
+
+const localProjectId: string = 'local'
+const clients: Record<string, PubSub> = {}
+
+const init = (config?: ClientConfig) => {
```
Comment (believer) [inline packages/pubsub/src/lib/client.ts:11]: We could simplify this with a default `config`:

```ts
const init = ({ projectId }: ClientConfig = { projectId: localProjectId }) => {
  if (!clients[projectId]) {
    if (projectId === localProjectId) {
      // Create a default client when there is no config.
      clients[projectId] = new PubSub()
    } else {
      clients[projectId] = new PubSub({
        projectId,
      })
    }
  }
}
```

File: packages/pubsub/src/lib/client.ts [hunk 2]
```diff
@@ -1,19 +1,48 @@
 import type { Topic } from '@google-cloud/pubsub'
 import { PubSub } from '@google-cloud/pubsub'
 
-export const pubsub = new PubSub()
+export type ClientConfig = {
+  projectId: string
+}
+
+const localProjectId: string = 'local'
```
Comment (believer) [inline packages/pubsub/src/lib/client.ts:8]: Remove `string` type, TypeScript infers it.

File: packages/pubsub/src/lib/client.ts [hunk 3]
```diff
@@ -1,19 +1,48 @@
...
+const clients: Record<string, PubSub> = {}
+
+const init = (config?: ClientConfig) => {
+  let projectId: string = localProjectId
```
Comment (believer) [inline packages/pubsub/src/lib/client.ts:12]: If we don't use the default `config` above, remove the `string` type here.

**Review by believer — APPROVED** [review]

File: packages/pubsub/src/lib/client.ts (hunk 1 above)
Comment (alexanderczigler) [reply packages/pubsub/src/lib/client.ts:11]: Great idea!

**Pull request merged**